forward kinetic to latest changes in devel

minhnh opened the pull request:
As discussed in #32, I and @mhwasil discussed and agreed that now is a good time for merging `devel` into `kinetic`. The changes in this PR include:
* various enhancements of the utilities in `mas_perception_libs`
* refactor point cloud filtering and plane fitting from `mcr_scene_segmentation` into separate definitions in `mas_perception_libs`
* resolve various build warnings, linting errors

Additionally, issues #4, #5, as well as documentation of the other packages in the form of `README` files, may need to be addressed before this merge. On another hand, if the scope of these issues is too big, we can address them in another PR.

minhnh committed:
add vis. functions and RGB detector base class

minhnh committed:
add detection service and detection test class

- test class returns random boxes
- add example kwargs file for test class
- add example class annotation file
- add BoundingBox2D message, ImageDetection.srv

minhnh committed:
add test script and launch file for detection

minhnh committed:
move loading kwargs file to base class

minhnh committed:
add boost python conversion for cv::Mat

- move draw bounding boxes to C++ to test cv::Mat conversion
- move generating detection message from prediction to separate method
- add DetectionServiceProxy sub class for image detection
- remove generate_msgs
- handle roslint errors

minhnh committed:
Merge branch 'refactor/move_perception_libs' of https://github.com/minhnh/mas_perception into feature/rgb_object_detection

minhnh committed:
replace enum with struct 2D for bounding boxes

minhnh committed:
expose fit_box_to_image and crop_image to python

- refactor cropImage to common/bounding_box_2d
- replace cv::Rect field with function
- add type check in visualization.py

minhnh committed:
add image detection from cloud and image topics

- add methods to extract image from organized clouds, expose them to Python
- add functionality to image_detection_client_test: detect objects from cloud/image topic
- add method ImageDetector.detection_msg_to_bounding_boxes()
- add visualization.draw_labeled_boxes_img_msg() to draw boxes on image messages

minhnh committed:
add cropping point cloud using 2D box

minhnh committed:
add extracting coordinates with bounding box

minhnh committed:
change launch namespace

minhnh committed:
add drawing labeled boxes for image messages

also move boost python related to _cpp_wrapper target for correct linking

minhnh committed:
add cloud transform python interface

minhnh committed:
update frame_id in transform_point_cloud()

minhnh committed:
Merge branch 'refactor/move_perception_libs' of https://github.com/minhnh/mas_perception into feature/rgb_object_detection

minhnh committed:
Merge branch 'refactor/move_perception_libs' of https://github.com/minhnh/mas_perception into feature/rgb_object_detection

minhnh committed:
refactor image recognition

minhnh committed:
add image detection object server

minhnh committed:
rewrite test script to share action server logic

minhnh committed:
remove messages added for detection service

minhnh committed:
Merge branch 'refactor/move_perception_libs' of https://github.com/minhnh/mas_perception into feature/rgb_object_detection

minhnh committed:
Merge branch 'refactor/move_perception_libs' of https://github.com/minhnh/mas_perception into feature/rgb_object_detection

minhnh committed:
refactor and add documentation

- refactor preprocessing image messages to base class for ImageDetector
- add comments and documentation

minhnh committed:
finalize documentation for C++ library

minhnh committed:
add in-code comments, restructure Python doc

minhnh committed:
Merge branch 'refactor/move_perception_libs' of https://github.com/minhnh/mas_perception into feature/rgb_object_detection

minhnh committed:
update Python documentation

minhnh committed:
remove unused detection test server

minhnh committed:
update README.md

minhnh committed:
Merge branch 'kinetic' of https://github.com/b-it-bots/mas_perception into feature/rgb_object_detection

minhnh committed:
add usage example

minhnh committed:
rename to ImageDetectorBase

minhnh committed:
refactor & update documentation

- change parameters to const ref as needed
- rename fields to match class name
- add intro to library in README

minhnh committed:
add image detection architecture

minhnh committed:
fix image link

minhnh committed:
remove pose estimation funtion

minhnh committed:
document modifying box size to fit cloud, images

minhnh committed:
Merge pull request #16 from minhnh/feature/rgb_object_detection

Feature/rgb object detection

argenos committed:
Add code owners file

argenos committed:
Merge branch 'kinetic' into devel

argenos committed:
Merge branch 'kinetic' into devel

minhnh committed:
refactor ROS independe point cloud functions

minhnh committed:
refactor cloud filtering and plane fitting from scene_segmentation

minhnh committed:
add dynamic config for cloud filtering and test node

minhnh committed:
refactor cloud conversion to point_cloud_utils_ros.cpp

minhnh committed:
add python interface and refactor cloud filtering

- move boost code to boost_python_module.cpp
- add test script and launch file for Python
- rename classes and files

minhnh committed:
refactor ROS cloud filtering and plane fitting into single class

minhnh committed:
add struct for fitted plane info, refactor to return plane messages

minhnh committed:
refactor mcr_perception_msgs/Plane and adapt code

minhnh committed:
add visualisation method, use correct SAC model

minhnh committed:
expose finding planes to Python

minhnh committed:
adapt object_detector module to new Plane message

minhnh committed:
change Plane message

minhnh committed:
change Plane message

minhnh committed:
expose plane message conversion to marker to Python

minhnh committed:
add dynamic param for passthrough on all 3 axes

minhnh committed:
remove redundant color.h, add Marker publish to Python cloud process script

minhnh committed:
use cloud filter and plane segment from mas_perception_libs

minhnh committed:
use c++11 for mcr_scene_segmentation, fix roslint errors

minhnh committed:
add plane x,y limit calculation, reolve deprecation warning

minhnh committed:
add fields for x, y coordinate ranges

minhnh committed:
Merge branch 'devel' of https://github.com/b-it-bots/mas_perception into refactor/change_plane_message

minhnh committed:
Merge branch 'refactor/change_plane_message' of https://github.com/minhnh/mas_perception into refactor/move_find_plane_function

minhnh committed:
add message for 2D bounding box and adapt Plane message

minhnh committed:
Merge branch 'refactor/change_plane_message' of https://github.com/minhnh/mas_perception into refactor/move_find_plane_function

minhnh committed:
adapt to new Plane message

alex-mitrevski committed:
Merge pull request #21 from minhnh/refactor/change_plane_message

Refactor/change plane message

minhnh committed:
add checking if object is on plane

minhnh committed:
use minimum x coord for object pose, add docs

minhnh committed:
remove redundant object pose calc in object_detector.py

minhnh committed:
skipp 3D boudning box calculations

minhnh committed:
add header to object pose

minhnh committed:
finalize documentation for plane segment. changes

minhnh committed:
Merge branch 'devel' of https://github.com/b-it-bots/mas_perception into refactor/move_find_plane_function

alex-mitrevski committed:
Merge pull request #19 from minhnh/refactor/move_find_plane_function

Refactor/move find plane function

minhnh committed:
improve handling plane detection errors

- return false in ObjectDetector.start_detect_objects if detection fails or aborted
- add timeout for detection action and waiting on cloud topic

alex-mitrevski committed:
Merge pull request #25 from minhnh/handle_plane_error

improve handling plane detection errors

argenos committed:
[travis-ci] Fix Travis stages

minhnh committed:
Merge branch 'devel' into kinetic

minhnh committed:
rename all ImageRecognition usage to RecognizeImage

argenos committed:
Merge pull request #32 from minhnh/kinetic

merge devel into kinetic

mhwasil commented on mcr_scene_segmentation/ros/config/scene_segmentation_constraints.yaml:
```diff
@@ -1,12 +1,12 @@
-    passthrough_filter_field_name: x
-    passthrough_filter_limit_min: 0.0
-    passthrough_filter_limit_max: 0.8
+    passthrough_limit_min_x: 0.0
```
As I mentioned in the prev PR, rename these params to avoid confusion (i.e. voxel_filter_limit...)

minhnh replied:
yeah that's reasonable, will do that later today

minhnh committed:
update dependencies, add initial test file

minhnh committed:
add unit test related utility functions

- add function to return relative path within a ROS package
- add function to return all messages of a certain type in a bag file
- add function to transform point clouds using rotation and translation
- remove type check for common type PointCloud2 to make it work with bag file types
- add tests for could to image conversion, cropping clouds, transforming clouds
- load test configuration from YAML file

minhnh committed:
add plane segmentation test, use quaternion for transforms

minhnh committed:
use bag file from mas_domestic_datasets

minhnh committed:
rename passthrough limit to voxel limit for z-axis

minhnh committed:
update dynamic config for mcr_scene_segmentation

minhnh committed:
add dependent data repo for mas_perception_libs

minhnh committed:
add Git LFS pull in Dockerfile

minhnh committed:
remove dep on mds_home_setups to avoid Travis fail

minhnh committed:
Merge pull request #35 from minhnh/refactor/change_dynamic_configure_name

[refactor] change dynamic config param name

minhnh commented:
@mhwasil @sthoduka @deebuls @alex-mitrevski please approve so we can merge before German Open (for a clean start).

minhnh commented:
For some reason `docker push` failed at the end of the build. I'm guessing this is an unrelated issue with DockerHub?

argenos commented:
Well, there is something fishy going on here. The build fails in `travis.com`, but not in `travis.org`, so it's probably something in the configuration now. I also had to re-add the environment variables for Docker Hub which were for some reason not there anymore.

argenos commented:
Seems to work now... I fixed a wrongly added environment variable and removed some old service configuration.
